### PR TITLE
Remove AigBitblaster implementation if ABC is not compiled

### PR DESCRIPTION
**Pull request by mpreiner**
Default implementation uses stubs with Unreachable() in every method + ctor + dtor. Remove default implementation and guard use of AigBitblaster with CVC4_USE_ABC.

### CODE REVIEW

**Comment by barrettcw** [comment]
Mathias - I agree that removing the default implementation with all of the Unreachable assertions is a good idea.  However, I don't like using #ifdef unless it absolutely can't be avoided.  My suggestion in this case would be to try to limit the use of #ifdef to maybe one line where the abc AIG bitblaster is created and otherwise can't we just check if the AIG bitblaster is NULL rather than using an #ifdef?

**Comment by mpreiner** [comment]
I think we can get it down to three #ifdef in  bv_eager_solver.cpp. One for creating AigBitblaster and the other two for guarding the calls to d_aigBitblaster->bbFormula() and d_aigBitblaster->solve().

File: src/theory/bv/bv_eager_solver.cpp [hunk 1]
```diff
@@ -34,26 +34,32 @@ EagerBitblastSolver::EagerBitblastSolver(TheoryBV* bv)
...
-          d_bitblaster == NULL);
+  Assert (d_aigBitblaster == NULL);
+  Assert (d_bitblaster == NULL);
   d_useAig = false;
 }
 
 void EagerBitblastSolver::initialize() {
   Assert(!isInitialized());
   if (d_useAig) {
+#ifdef CVC4_USE_ABC
```
Comment (4tXJ7f) [inline src/theory/bv/bv_eager_solver.cpp]: This `#ifdef` is a bit inconsistent in style with the other ones. I like the other style better (where you have the `#ifdef` inside of the `then` block.

**Comment by barrettcw** [comment]
Yes, this is better.  Thanks.

File: src/theory/bv/bv_eager_solver.cpp (hunk 1 above)
Comment (4tXJ7f) [inline src/theory/bv/bv_eager_solver.cpp]: The issue was actually that the diff showed up a bit weird, which confused me. All is well, my bad.

**Pull request merged**